src/Services/FHIR/FhirExportJobService.php

```diff
@@ -0,0 +1,161 @@
+        $ret = sqlQueryNoLog($sql, $params);
+
+        if (empty($ret)) {
+            $this->logger->error(
+                "FhirExportRestController->getJobForId() failed to find job",
+                ['jobUuid' => $jobUuidString, 'sql' => $sql, 'params' => $params]
+            );
+            throw new \InvalidArgumentException("Export Job with jobId '" . $jobUuidString . "' does not exist");
+        }
+
+        $this->logger->debug("FhirExportRestController->getJobForId() ", ['jobId' => $jobUuidString, 'dbResult' => $ret]);
+
+        $job = new ExportJob();
+        $job->setId($ret['id']);
+        $job->setUuid($jobUuid);
+        $job->setStartTime(\DateTime::createFromFormat(self::DATETIME_FORMAT, $ret['start_time']));
+        $job->setResourceIncludeTime(\DateTime::createFromFormat(self::DATETIME_FORMAT, $ret['resource_include_time']));
+        $job->setOutputFormat($ret['output_format']);
+        $job->setResources($ret['resources']);
+        $job->setClientId($ret['client_id']);
+        $job->setUserId($ret['user_id']);
+        $job->setAccessTokenId($ret['access_token_id']);
+        $job->setStatus($ret['status']);
+        $job->setRequestURI($ret['request_uri']);
+        $job->setOutput($ret['output']);
+        $job->setErrors($ret['errors']);
+        return $job;
+    }
+
+    /**
+     * Given an export job, save it to the database
+     * @param ExportJob $job The job to save
+     * @return ExportJob the saved job
+     * @throws \RuntimeException if the job fails to save
+     */
+    public function createJobRequest(ExportJob $job)
+    {
+        // we will generate a UUID here, if we ever want the db to do that we would accomplish that here...
+        $binaryUuid = (new UuidRegistry(['table_name' => 'export_job']))->createUuid();
+        $job->setUuid($binaryUuid);
+
+        $sql = "INSERT INTO `export_job`(`uuid`, `start_time`, `resource_include_time`, `output_format`, `resources`, "
+            . "`client_id`, `user_id`, `access_token_id`, `status`, `request_uri`) "
+            . " VALUES (?,?,?,?,?,?,?,?,?,?)";
+
+        $startTime = $job->getStartTime()->format(self::DATETIME_FORMAT);
+        if ($job->getResourceIncludeTime() instanceof \DateTime) {
+            $resourceIncludeTime = $job->getResourceIncludeTime()->format(self::DATETIME_FORMAT);
+        } else {
+            $resourceIncludeTime = null;
+        }
+        $params = [$job->getUuid(), $startTime, $resourceIncludeTime
+            , $job->getOutputFormat(), $job->getResourcesString(), $job->getClientId(), $job->getUserId()
+            , $job->getAccessTokenId(), $job->getStatus(), $job->getRequestURI()];
+        // TODO: @bradymiller is there a different function we can call that will NOT kill execution if the insert fails?
+        // for an API caller we'd rather throw an exception or something else rather than just dying so we can communicate
+        // back to the caller that something went wrong.
```

Comment:
we could make a sqlInsertThrowError function that would instead throw an error rather than die

Reply:
I went ahead and made that function.